### ISSUE TITLE
Add unit tests for api-tools/parser/validator

### ISSUE THREAD

**Issue by jpahm** [UTDNebula/api-tools#41]
Add unit tests using [Go's built-in testing library](https://pkg.go.dev/testing). I would personally be in favor of the test files being within the same package as the files being tested so that non-exported methods can be tested more easily; make sure to follow the `_test.go` file naming convention detailed in the linked docs. **Be sure to seek out resources for details on how to write good unit tests if you are unsure.**

**Comment by mikehquan19** [comment]
Can I work on this ?

**Comment by jpahm** [comment]
> Can I work on this ?

Sure thing!

**Comment by jpahm** [comment]
@mikehquan19 How's this going?

**Comment by mikehquan19** [comment]
Yes I'm still working on this. I will go ahead and create PR on this once I've finished all of my aggregate endpoint tasks. 
Thank you.

**Comment by mikehquan19** [comment]
I have a few questions regarding this.
1/ Since there is only one function in `validator.go`, do we have multiple tests for 1 function, each testing one part of the function, or 1 test for the whole func ?

2/ How can I get the test data of `Courses`, `Professors`, & `Sections` to test the function? Considering the fact that they validate directly the parsed data. I'm thinking about breaking the function into smaller function validating smaller part. From there we can write unit tests better.

**Comment by jpahm** [comment]
Hi, apologies, I missed this comment!
>Since there is only one function in validator.go, do we have multiple tests for 1 function, each testing one part of the function, or 1 test for the whole func?

Generally you'll want to have as many tests as needed to cover appropriate checks; this will be up to the test designer's discretion. That being said, Go's built-in testing library supports subtests.

>How can I get the test data of Courses, Professors, & Sections to test the function? Considering the fact that they validate directly the parsed data. I'm thinking about breaking the function into smaller function validating smaller part. From there we can write unit tests better.

Breaking into smaller chunks like this is exactly what you want to do for unit tests, so you're on the right track! As far as test data goes, one option would be to take a known set of inputs and associated outputs. We could also look into using the built-in test fuzzing feature.

**Comment by mikehquan19** [comment]
Related to PR #63
